bentoml/_internal/utils/buildx.py

```diff
@@ -0,0 +1,303 @@
+    shm_size: str | int | None,
+    rm: bool,
+    ssh: str | None,
+    tags: str | list[str] | None,
+    target: str | None,
+    ulimit: str | None,
+) -> None:
+    cmds = DOCKER_BUILDX_CMD + ["build"]
+
+    cmds += ["--progress", progress]
+
+    if tags is None:
+        tags = []
+    tags = [tags] if not isinstance(tags, list) else tags
+    for tag in tags:
+        cmds.extend(["--tag", tag])
+
+    if add_host is not None:
+        hosts = [f"{k}:{v}" for k, v in add_host.items()]
+        for host in hosts:
+            cmds.extend(["--add-host", host])
+
+    if allow is not None:
+        for allow_arg in allow:
+            cmds.extend(["--allow", allow_arg])
+
+    if build_args is not None:
+        args = [f"{k}={v}" for k, v in build_args.items()]
+        for arg in args:
+            cmds.extend(["--build-arg", arg])
+
+    if build_context is not None:
+        args = [f"{k}={v}" for k, v in build_context.items()]
+        for arg in args:
+            cmds.extend(["--build-context", arg])
+
+    if builder is not None:
+        cmds.extend(["--builder", builder])
+
+    if cache_from is not None:
+        if isinstance(cache_from, str):
+            cmds.extend(["--cache-from", cache_from])
+        elif isinstance(cache_from, list):
+            for arg in cache_from:
+                cmds.extend(["--cache-from", arg])
+        else:
+            args = [f"{k}={v}" for k, v in cache_from.items()]
+            cmds.extend(["--cache-from", ",".join(args)])
+
+    if cache_to is not None:
+        if isinstance(cache_to, str):
+            cmds.extend(["--cache-to", cache_to])
+        elif isinstance(cache_to, list):
+            for arg in cache_to:
+                cmds.extend(["--cache-to", arg])
+        else:
+            args = [f"{k}={v}" for k, v in cache_to.items()]
+            cmds.extend(["--cache-to", ",".join(args)])
+
+    if cgroup_parent is not None:
+        cmds.extend(["--cgroup-parent", cgroup_parent])
+
+    if file is not None:
+        cmds.extend(["--file", str(file)])
+
+    if iidfile is not None:
+        cmds.extend(["--iidfile", str(iidfile)])
+
+    if load:
+        cmds.append("--load")
+
+    if metadata_file is not None:
+        cmds.extend(["--metadata-file", str(metadata_file)])
+
+    if network is not None:
+        cmds.extend(["--network", network])
+
+    if no_cache:
+        cmds.append("--no-cache")
+
+    if no_cache_filter is not None:
+        for arg in no_cache_filter:
+            cmds.extend(["--no-cache-filter", arg])
+
+    if labels is not None:
+        args = [f"{k}={v}" for k, v in labels.items()]
+        for arg in args:
+            cmds.extend(["--label", arg])
+
+    if output is not None:
+        if isinstance(output, str):
+            cmds.extend(["--output", output])
+        else:
+            args = [f"{k}={v}" for k, v in output.items()]
+            cmds += ["--output", ",".join(args)]
+
+    if platform is not None:
```

Comment:
`""` will cause a failure

Reply:
have a fixed on the docker revamp branch